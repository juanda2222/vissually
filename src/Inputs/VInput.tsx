import React, {
  useContext,
  useState,
  //useEffect,
  //useRef,
} from 'react'


import styles from './VInput.module.css'
import styled from 'styled-components';
import { Color2Vec } from "../Tools/ColorTools"
import { DefaultThemes, ThemeContext } from "../ThemeProvider/ThemeProvider"


const StyledInput = styled("input") <{ rgbTextColor: number[], rgbFocusedColor: number[] }>`
  border: 2px solid ${props => `rgb(${props.rgbTextColor[0] -30}, ${props.rgbTextColor[1] -30}, ${props.rgbTextColor[2] -30})`};
  color: ${props => `rgb(${props.rgbTextColor[0]}, ${props.rgbTextColor[1]}, ${props.rgbTextColor[2]})`};
  &:focus {
    border-color: ${props => `rgb(${props.rgbFocusedColor[0]}, ${props.rgbFocusedColor[1]}, ${props.rgbFocusedColor[2]});`};
  }
   
`;

const StyledLabel = styled("label") <{
  isFocused: boolean,
  rgbTextColor: number[],
  rgbFocusedTextColor: number[]
}>`
  color: ${props => props.isFocused ?
    `rgb(${props.rgbFocusedTextColor[0]}, ${props.rgbFocusedTextColor[1]}, ${props.rgbFocusedTextColor[2]})` :
    `rgb(${props.rgbTextColor[0]}, ${props.rgbTextColor[1]}, ${props.rgbTextColor[2]})`
  };
`;

function VTextInput ({
  //styling properties
  className,
  style,
  parentStyle,
  labelStyle,
  theme,
  primary, secondary, primaryDark, secondaryDark,

  //functional properties
  value,
  id,
  name,
  type,
  required,
  placeholder,
  onChange,
  onFocus,
  onBlur,
}: InputProps ){  

  // get the theme from the context
  const context_theme = useContext(ThemeContext);
  const [inputIsFocused, setInputFocused] = useState(false)

  // all the event handlers wrapped for custom styling
  const wrappedOnFocus = (
    event: React.FocusEvent<HTMLInputElement>,
  ) => {
    setInputFocused(true)
    if (typeof onFocus === 'function') {onFocus(event)}
  }

  const wrappedOnBlur = (
    event: React.FocusEvent<HTMLInputElement>,
  ) => {
    setInputFocused(false)
    if (typeof onBlur === 'function') {onBlur(event)}
  }

  // generate the theme depending on the boolean inputs or the theme input
  const wrappedClassName = [styles.input_element, className ? className : ""].join(" ")
  let current_theme: DefaultTheme;
  if (secondary) {
    current_theme = DefaultThemes.secondary
  } else if (primaryDark) {
    current_theme = DefaultThemes.primaryDark
  } else if (secondaryDark) {
    current_theme = DefaultThemes.secondaryDark
  } else if (primary) {
    current_theme = DefaultThemes.primary
  } else if (!(typeof (context_theme) == "undefined")) {
    current_theme = context_theme
  } else {
    current_theme = {
      ...DefaultThemes.primary,
      ...(theme && theme)
    } 
  }
  
  // generate the component from the style
  const rgbMainColor = Color2Vec(current_theme.color1)
  const rgbTextColor = Color2Vec(current_theme.textColor1)
  
  return (
    <div
      className={styles.container}
      style={parentStyle}
    >
      <StyledInput
        //custom style:
        rgbFocusedColor={rgbMainColor}
        rgbTextColor={rgbTextColor}
        // parent styling props:
        className={wrappedClassName}
        style={style}
        // functional props:
        value={value}
        placeholder={placeholder}
        id={id}
        name={name}
        type={type}
        required={required}
        onChange={onChange}
        onFocus={wrappedOnFocus}
        onBlur={wrappedOnBlur}
      />      
      <StyledLabel
        className={styles.label_element}
        style={labelStyle}
        isFocused={inputIsFocused}
        rgbTextColor={rgbTextColor}
        rgbFocusedTextColor={rgbMainColor}
        
      >
        {placeholder}
      </StyledLabel>
    </div>
  )
}


export default VTextInput