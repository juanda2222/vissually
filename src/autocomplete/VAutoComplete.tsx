import React, { useContext, useState, useEffect, useRef } from 'react'


import styles from './VAutoComplete.module.css'
import { DefaultThemes, ThemeContext } from "../ThemeProvider/ThemeProvider"
import VButton from "../Buttons/VButton"
import VClickableList from "../Lists/VClickableList"
import ArrowDownSVG from "../assets/arrow-down.js"
import ArrowUpSVG from "../assets/arrow-up.js"


const VAuto: React.FunctionComponent<AutoCompleteProps> = ({
  //functional properties
  id,
  options,
  getOptionLabel,
  style,
  renderInput,

}) => {

  return (
    <div
      ref={wrapperRef}
      className={styles.big_container}>
      <VButton
        isPressed={isOpen}
        style={style}
        className={wrapped_className}
        theme={current_theme}
        onClick={() => {
          setOpen(!isOpen)
        }}
      >
        
        {selectedItem == "" ? label : selectedItem}
        
        {isOpen ?
          <ArrowUpSVG style={{
            fill: current_theme.textColor2,
            width: "0.7em", height: "0.7em",
            margin: "auto 6px"
          }} /> :
          <ArrowDownSVG style={{
            fill: current_theme.textColor2,
            width: "0.7em", height: "0.7em",
            margin: "auto 6px"
          }} />
        }

      </VButton>
      <VClickableList
        list={list}
        className={styles.list_item}
        containerClassName={wrapped_list_className}
        onClick={(index, value) => {
          setOpen(false)
          setSelectedItem(value)
          onSelect && onSelect(index, value)
        }}
      />
    </div>
  )
}


export default VSelect