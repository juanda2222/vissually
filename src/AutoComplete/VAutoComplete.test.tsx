
import React from "react"
import renderer from 'react-test-renderer';
//import ReactDOM from 'react-dom';
import VAutoComplete from './VAutoComplete'
import VInput from '../Inputs/VInput'

const autocomplete_list = [
  { title: 'The Shawshank Redemption', year: 1994 },
  { title: 'The Godfather', year: 1972 },
  { title: 'The Godfather: Part II', year: 1974 },
  { title: 'The Dark Knight', year: 2008 },
  { title: '12 Angry Men', year: 1957 },
  { title: "Schindler's List", year: 1993 },
]

describe('AutoComplete', () => {

  test('AutoComplete declaration has no errors', () => {
    expect(VAutoComplete).toBeTruthy()
  })

  test("Default render. Snapshot test", () => { 
    const AutoCompleteTree = renderer.create(<VAutoComplete/>).toJSON();
    console.log(AutoCompleteTree)
    expect(AutoCompleteTree).toMatchSnapshot();
  })

  test("Rendering with props. Snapshot test", () => { 
    
    const AutoCompletePropsTree = renderer.create(
      <VAutoComplete
        options={autocomplete_list}
        getOptionLabel={(option: { [prop:string]: any }) => option.title}
        style={{ width: 300 }}
        renderInput={(params: { [prop:string]: any }) => <VInput {...params} placeholder="Combo box" />}
      />
    ).toJSON();

    console.log(AutoCompletePropsTree)
    expect(AutoCompletePropsTree).toMatchSnapshot();
  })


})

