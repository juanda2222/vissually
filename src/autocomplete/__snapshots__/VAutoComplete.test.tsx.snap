// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete Default render. Snapshot test 1`] = `
<div
  className="big_container"
/>
`;

exports[`AutoComplete Rendering with props. Snapshot test 1`] = `
<div
  className="big_container"
>
  <div
    className="container"
  >
    <input
      className="sc-AxheI kbekTD input_element input_style"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Combo box"
      value=""
    />
    <label
      className="sc-Axmtr ixJmyd label_element"
    >
      Combo box
    </label>
  </div>
  <div
    className="sc-AxjAm jNAsBD list_container list_style fadeOut "
  >
    <div
      className="sc-AxirZ sc-AxiKw crVGut list_item "
      onClick={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      The Shawshank Redemption
    </div>
    <div
      className="sc-AxirZ jFyzcN list_item "
      onClick={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      The Godfather
    </div>
    <div
      className="sc-AxirZ jFyzcN list_item "
      onClick={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      The Godfather: Part II
    </div>
    <div
      className="sc-AxirZ jFyzcN list_item "
      onClick={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      The Dark Knight
    </div>
    <div
      className="sc-AxirZ jFyzcN list_item "
      onClick={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      12 Angry Men
    </div>
    <div
      className="sc-AxirZ sc-AxhCb fvJznW list_item "
      onClick={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      Schindler's List
    </div>
  </div>
</div>
`;
